src/helpers/common.ts

```diff
@@ -105,3 +135,56 @@ export function isNullish<T>(
 ): value is null | undefined {
   return value === null || value === undefined;
 }
+
+export function callbackifyGenerator<T>(generatorFn: () => AsyncGenerator<T>) {
+  const generator = generatorFn();
+  return (callback: AnyFn) => {
+    (async () => {
+      try {
+        for await (const result of generator) {
+          callback(null, result);
+        }
+      } catch (err) {
+        callback(err);
+      }
+    })();
+    return;
+  };
+}
+
+export async function* paginator<T>(
+  executor: (searchParams: URLSearchParams) => Promise<{
+    results: T[];
+    totalCount: number;
+  }>,
+  {
+    offset = 0,
+    count = Infinity,
+    params,
+    limit = 100,
+  }: {
+    offset?: number;
+    count?: number;
+    params?: URLSearchParams;
+    limit?: number;
+  },
+): AsyncGenerator<T> {
+  let currentOffset = offset;
+  let remainingCount = count;
+  let totalCount = Infinity;
+  while (currentOffset < totalCount) {
+    const paginatedSearchParams = new URLSearchParams(params);
+    paginatedSearchParams.set('offset', currentOffset.toString());
+    paginatedSearchParams.set(
+      'limit',
+      Math.min(remainingCount, limit).toString(),
+    );
+    const output = await executor(paginatedSearchParams);
+    for (const result of output.results) {
+      yield result;
+      if (--remainingCount === 0) return;
+      ++currentOffset;
+    }
+    totalCount = output.totalCount;
+  }
+}
```

Comment:
Can we give the client information about the total amount, as we have this information from the API?